cookbooks/smoke-tests/recipes/oozie_smoke_test.rb

```diff
@@ -92,18 +97,33 @@
     :user => test_user
 end
 
+execute "upload hive_smoke_test to #{workflow_path}" do
+  command "hdfs dfs -copyFromLocal -f #{cache_dir}/hive_smoke_test.hql \
+  #{workflow_path}"
+  user test_user
+  not_if "hdfs dfs -test -f #{workflow_path}/hive_smoke_test.hql",
+         :user => test_user
+end
+
+execute "upload hive-site to #{workflow_path}" do
+  command "hdfs dfs -copyFromLocal -f /etc/hive/conf/hive-site.xml \
```

Comment:
I understand `copyFromLocal` to fail if he file already exists? How is this made idempotent?

Reply:
the `-f` option overrides the existing file.

Reply:
Cool! Then should line 104's `not_if` continue to exist?